js/foam/ui/md/Halo.js

```diff
@@ -151,11 +155,21 @@ CLASS({
           this.x = evt.offsetX;
           this.y = evt.offsetY;
         }
-        this.r = 2;
+        if ( this.animateGrowth ) {
+          this.r = 2;
+        } else if ( this.recentering ) {
+          this.x = this.parent.width/2;
+          this.y = this.parent.height/2;
+          this.r = Math.min(28, Math.min(this.$.clientWidth, this.parent.height)/2);
+        } else {
+          this.r = Math.max(28, Math.max(this.$.clientWidth, this.parent.height));
+        }
         this.alpha = this.startAlpha;
-        var recentering = this.recentering;
+        const recentering = this.recentering;
         this.X.animate(this.easeInTime, function() {
-          if ( recentering ) {
+          if ( !this.animateGrowth ) {
+            // Do nothing.
```

Comment:
Why not

```
if (!this.animateGrowth) return;

if (recentering) {

} else {

}
```

Reply:
We still need line 179 which sets the alpha.

Reply:
Why not just move line 179 first and then do the if-else afterwards?

Reply:
Done!